tensorflow_probability/python/experimental/mcmc/sample_fold.py

```diff
@@ -133,3 +139,174 @@ def sample_fold(
     return (reduction_results,
             end_state,
             final_kernel_results.inner_results.inner_results)
+
+
+class StatesAndTrace(
+    mcmc_util.PrettyNamedTupleMixin,
+    collections.namedtuple('StatesAndTrace', ['all_states', 'trace'])):
+  """States and auxiliary trace of an MCMC chain.
+
+  The first dimension of all the `Tensor`s in this structure is the same and
+  represents the chain length.
+
+  Attributes:
+    all_states: A `Tensor` or a nested collection of `Tensor`s representing the
+      MCMC chain state.
+    trace: A `Tensor` or a nested collection of `Tensor`s representing the
+      auxiliary values traced alongside the chain.
+  """
+  __slots__ = ()
+
+
+class CheckpointableStatesAndTrace(
+    mcmc_util.PrettyNamedTupleMixin,
+    collections.namedtuple('CheckpointableStatesAndTrace',
+                           ['all_states', 'trace', 'final_kernel_results'])):
+  """States and auxiliary trace of an MCMC chain.
+
+  The first dimension of all the `Tensor`s in the `all_states` and `trace`
+  attributes is the same and represents the chain length.
+
+  Attributes:
+    all_states: A `Tensor` or a nested collection of `Tensor`s representing the
+      MCMC chain state.
+    trace: A `Tensor` or a nested collection of `Tensor`s representing the
+      auxiliary values traced alongside the chain.
+    final_kernel_results: A `Tensor` or a nested collection of `Tensor`s
+      representing the final value of the auxiliary state of the
+      `TransitionKernel` that generated this chain.
+  """
+  __slots__ = ()
+
+
+def sample_chain(
+    num_results,
+    current_state,
+    previous_kernel_results=None,
+    kernel=None,
+    num_burnin_steps=0,
+    num_steps_between_results=0,
+    trace_fn=lambda current_state, kernel_results: kernel_results,
+    return_final_kernel_results=False,
+    parallel_iterations=10,
+    seed=None,
+    name=None,
+):
+  """Implements Markov chain Monte Carlo via repeated `TransitionKernel` steps.
+
+  This function samples from an Markov chain at `current_state` and whose
+  stationary distribution is governed by the supplied `TransitionKernel`
+  instance (`kernel`).
+
+  This function can sample from multiple chains, in parallel. (Whether or not
+  there are multiple chains is dictated by the `kernel`.)
```

Comment:
What does this parenthetical comment mean?

Reply:
I was actually confused about this too... it's in the Docstring of `sample_chain` so I assumed it applied to some other kernels in mcmc that I haven't looked at in depth